From 6cb9741f158874c8110164b359aeb825637d14b2 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Bart=C5=82omiej=20Piotrowski?= <b@bpiotrowski.pl>
Date: Wed, 7 Jun 2023 12:27:16 +0200
Subject: [PATCH] Don't specify --components at all

---
 src/builder-manifest.c | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/src/builder-manifest.c b/src/builder-manifest.c
index 3cbbdff2..2ae36fc1 100644
--- a/src/builder-manifest.c
+++ b/src/builder-manifest.c
@@ -2769,8 +2769,6 @@ builder_manifest_cleanup (BuilderManifest *self,
         {
           g_autofree char *origin = g_strdup_printf ("--origin=%s",
                                                      builder_manifest_get_id (self));
-          g_autofree char *components_arg = g_strdup_printf ("--components=%s,%s.desktop",
-                                                             self->id, self->id);
           const char *app_root_path = flatpak_file_get_path_cached (app_root);
           g_autofree char *result_root_arg = g_strdup_printf ("--result-root=%s", app_root_path);
           g_autoptr(GFile) xml_dir = flatpak_build_file (app_root, "share/app-info/xmls", NULL);
@@ -2800,7 +2798,6 @@ builder_manifest_cleanup (BuilderManifest *self,
                                          result_root_arg,
                                          data_dir,
                                          icon_dir,
-                                         components_arg,
                                          app_root_path,
                                          NULL))
               return FALSE;
@@ -2814,7 +2811,6 @@ builder_manifest_cleanup (BuilderManifest *self,
                                          result_root_arg,
                                          data_dir,
                                          icon_dir,
-                                         components_arg,
                                          app_root_path,
                                          NULL))
                 return FALSE;
-- 
2.40.1

