diff --git a/src/as-validator.c b/src/as-validator.c
index 2576b149..7caf3fe0 100644
--- a/src/as-validator.c
+++ b/src/as-validator.c
@@ -659,6 +659,21 @@ as_validator_add_override (AsValidator *validator,
 		"developer-id-missing",
 		/* allow in case a component really doesn't have an active homepage */
 		"url-homepage-missing",
+		/* flathub apps often instruct users to follow some additional steps for sandbox-breaking apps */
+		"description-has-plaintext-url",
+		/* we have our own validation for these fields */
+		"component-name-too-long",
+		"summary-too-long",
+		/* this is a breaking change as 1.0.0 allowed this field to be anything */
+		"developer-id-invalid",
+		/* this has not been the case with appstream-glib */
+		"cid-domain-not-lowercase",
+		/* This check is broken because it triggers when desktop cpts
+		mistakenly use a stock icon in the metainfo. We have our
+		own check to ensure desktop cpts have category in catalogue
+		See: https://discourse.flathub.org/t/app-categories-missing-validation-error/8636/5
+		*/
+		"app-categories-missing",
 		NULL
 	};
 
