From 30c9e5e322da71e5b955cde43e07e60da92ee2af Mon Sep 17 00:00:00 2001
From: Richard Hughes <richard@hughsie.com>
Date: Sat, 8 Oct 2022 09:06:13 +0100
Subject: [PATCH] trivial: Do not expect screenshots from console apps

---
 libappstream-glib/as-app-validate.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/libappstream-glib/as-app-validate.c b/libappstream-glib/as-app-validate.c
index 85f4e45..a11c361 100644
--- a/libappstream-glib/as-app-validate.c
+++ b/libappstream-glib/as-app-validate.c
@@ -790,6 +790,7 @@ as_app_validate_screenshots (AsApp *app, AsAppValidateHelper *helper)
 	if (as_app_get_kind (app) == AS_APP_KIND_FIRMWARE ||
 	    as_app_get_kind (app) == AS_APP_KIND_DRIVER ||
 	    as_app_get_kind (app) == AS_APP_KIND_RUNTIME ||
+	    as_app_get_kind (app) == AS_APP_KIND_CONSOLE ||
 	    as_app_get_kind (app) == AS_APP_KIND_ADDON ||
 	    as_app_get_kind (app) == AS_APP_KIND_LOCALIZATION)
 		number_screenshots_min = 0;
-- 
2.37.3

