diff --git a/client/as-util.c b/client/as-util.c
index ce973ed..937440f 100644
--- a/client/as-util.c
+++ b/client/as-util.c
@@ -3313,19 +3313,6 @@ as_util_mirror_screenshots_app_file (AsApp *app,
 		g_debug ("%s is not in 16:9 aspect ratio", filename_no_path);
 	}
 
-	/* check screenshot is reasonable in size */
-	if (as_image_get_width (im_src) * 2 < AS_IMAGE_NORMAL_WIDTH ||
-	    as_image_get_height (im_src) * 2 < AS_IMAGE_NORMAL_HEIGHT) {
-		g_set_error (error,
-			     AS_APP_ERROR,
-			     AS_APP_ERROR_FAILED,
-			     "%s is too small to be used: %ux%u",
-			     filename_no_path,
-			     as_image_get_width (im_src),
-			     as_image_get_height (im_src));
-		return FALSE;
-	}
-
 	/* check the image is not padded */
 	alpha_flags = as_image_get_alpha_flags (im_src);
 	if ((alpha_flags & AS_IMAGE_ALPHA_FLAG_TOP) > 0||
