diff --git a/src/builder-main.c b/src/builder-main.c
index 42f4784e..cc3da354 100644
--- a/src/builder-main.c
+++ b/src/builder-main.c
@@ -153,6 +153,7 @@ static GOptionEntry run_entries[] = {
   { "log-session-bus", 0, 0, G_OPTION_ARG_NONE, &opt_log_session_bus, N_("Log session bus calls"), NULL },
   { "log-system-bus", 0, 0, G_OPTION_ARG_NONE, &opt_log_system_bus, N_("Log system bus calls"), NULL },
   { "ccache", 0, 0, G_OPTION_ARG_NONE, &opt_ccache, "Use ccache", NULL },
+  { "disable-rofiles-fuse", 0, 0, G_OPTION_ARG_NONE, &opt_disable_rofiles, "Disable rofiles-fuse use", NULL },
   { NULL }
 };
 
